pebble/pool/channel.py

```diff
@@ -43,7 +43,11 @@ def __init__(self, reader, writer):
     def _make_poll_method(self):
         def unix_poll(timeout=None):
             poll = select.poll()
-            poll.register(self.reader)
+            poll.register(self.reader, READ_ONLY_EVENTMASK)
+    
+            # Convert from Seconds to Milliseconds
+            timeout = timeout * 1000 if timeout else timeout
```

Comment:
It is usually an antipattern to test for None objects treating them as booleans:
https://www.python.org/dev/peps/pep-0008/#programming-recommendations

In this case, for example, if by chance the `unix_poll()` function would be given a large string, you would end up multiplying it by a factor of 1000 most likely sending the program to OOM.

It is always recommended to test explicitly against `None` values.

```python
timeout = timeout * 1000 if timeout is not None else None
```

Or more simply in this case:
```python
if timeout is not None:
    timeout *= MILLISECONDS

...

MILLISECONDS = 1000
```